ddtrace/internal/coverage/code.py

```diff
@@ -191,54 +291,96 @@ def report_seen_lines(cls):
         if cls._instance is None:
             return []
         files = []
-        covered = cls._instance._get_covered_lines()
+        covered = cls._instance._get_covered_lines(include_imported=include_imported)
+
+        for abs_path_str, lines in covered.items():
+            abs_path = Path(abs_path_str)
+            path_str = (
+                str(abs_path.relative_to(workspace_path)) if abs_path.is_relative_to(workspace_path) else abs_path_str
+            )
 
-        for path, lines in covered.items():
             sorted_lines = sorted(lines)
+
             collapsed_ranges = collapse_ranges(sorted_lines)
             file_segments = []
             for file_segment in collapsed_ranges:
                 file_segments.append([file_segment[0], 0, file_segment[1], 0, -1])
-            files.append({"filename": path, "segments": file_segments})
+            files.append({"filename": path_str, "segments": file_segments})
 
         return files
 
     def transform(self, code: CodeType, _module: ModuleType) -> CodeType:
         code_path = Path(code.co_filename)
+        if _module is None:
+            return code
 
         if not any(code_path.is_relative_to(include_path) for include_path in self._include_paths):
             # Not a code object we want to instrument
             return code
 
-        return self.instrument_code(code)
+        if any(code_path.is_relative_to(_) for _ in (stdlib_path, purelib_path)):
```

Comment:
I think the logic for detecting stdlib might be more complicated because of how things are installed https://github.com/P403n1x87/dd-trace-py/blob/c81984899843c776254ed30d35de232d87d3e915/ddtrace/internal/packages.py#L226